vcs/gitcmd/repo.go

```diff
@@ -150,23 +150,55 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-func (r *Repository) Branches() ([]*vcs.Branch, error) {
+// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
+func (r *Repository) branchCounts(branch, base string) (behind, ahead int, err error) {
+	cmd := exec.Command("git", "rev-list", "--count", "--left-right", base+"..."+branch)
```

Comment:
From my (can't guarantee to be thorough) testing, it's possible to create weird branch names, but not ones that start with a dash.

However, you may have a branch named `dev` and another branch literally named `refs/heads/dev`. If you use the short branch name in the comparison, `git` prints a warning:

```
$ git rev-list --count --left-right refs/heads/master...refs/heads/dev
warning: refname 'refs/heads/dev' is ambiguous.
0	1
```

But if you prefix the branch name with "refs/heads/", it's no longer ambiguous:

```
$ git rev-list --count --left-right refs/heads/master...refs/heads/refs/heads/dev
27	5
```

But then you are allowed to make a branch literally named "refs/heads/refs/heads/dev", which makes the above ambiguous again.